src/Domain/PlanningUpdateDomain.php

```
@@ -0,0 +1,210 @@
+
+    protected ObjectManager $om;
+    protected string $action;
+    protected array $payload;
+    protected User $planningAgent;
+
+    public function __construct(ObjectManager $om, string $action, array $payload, User $planningAgent)
+    {
+        $this->om = $om;
+        $this->action = $action;
+        $this->payload = $payload;
+        $this->planningAgent = $planningAgent;
+    }
+
+    public function compute(): void
+    {
+        $this->validateAction();
+        $this->validatePayload();
+
+        if (isset($this->payload[self::PAYLOAD_USERS_KEY]) && is_array($this->payload[self::PAYLOAD_USERS_KEY])) {
+            $this->process(User::class, UserAvailability::class, $this->payload[self::PAYLOAD_USERS_KEY]);
+        }
+
+        if (isset($this->payload[self::PAYLOAD_COMMISSIONABLE_ASSETS_KEY]) && is_array($this->payload[self::PAYLOAD_COMMISSIONABLE_ASSETS_KEY])) {
+            $this->process(CommissionableAsset::class, CommissionableAssetAvailability::class, $this->payload[self::PAYLOAD_COMMISSIONABLE_ASSETS_KEY]);
+        }
+
+        $this->om->flush();
+    }
+
+    protected function process($entityClass, $availabilityClass, array $data)
+    {
+        $entityRepository = $this->om->getRepository($entityClass);
+        $availabilityRepository = $this->om->getRepository($availabilityClass);
+
+        list($startSearch, $endSearch) = $this->getSearchDates($data);
+
+        $entities = $entityRepository->findBy(['id' => array_keys($data)]);
+        $availabilities = $availabilityRepository->findByOwnerAndDates($entities, new \DateTimeImmutable($startSearch), new \DateTimeImmutable($endSearch));
+
+        foreach ($data as $entityId => $schedules) {
+            $search = array_filter($entities, fn ($entity) => $entityId == $entity->id);
+            /** @var AvailabilityInterface */
+            $currentEntity = end($search);
+
+            if (!$currentEntity) {
+                throw new BadRequestHttpException('Invalid entity');
```

`BadRequestHttpException` should only be used in a controller

---

you can throw a `\InvalidArgumentException` or a `RuntimeException`